src/Core/Parser/SyntaxTree/Expression/MathExpressionNode.php

```diff
@@ -55,6 +55,8 @@ public static function evaluateExpression(RenderingContextInterface $renderingCo
                 $operator = $part;
             } else {
                 $part = static::getTemplateVariableOrValueItself($part, $renderingContext);
+                $part = ($part == (int)$part) ? (int)$part : (float)$part;
+                $result = ($result == (int)$result) ? (int)$result : (float)$result;
```

Comment:
I think it should be enough to do this once right after line 50. Or am I missing something?